homework/src/index.js

```diff
@@ -31,14 +30,92 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+    const root = document.getElementById('root');
+    const div = createAndAppend('div', root, { class: 'data' });
+    const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+    const table = createAndAppend('table', root, { class: 'table' });
+    const select = createAndAppend('select', header, { id: 'list' });
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+    const repositoryLink = createAndAppend('a', tr);
+    const description = createAndAppend('tr', tbody, { class: 'label' });
+    const forks = createAndAppend('tr', tbody, { class: 'label' });
+    const container = createAndAppend('div', root, { id: 'container' });
+    const updated = createAndAppend('tr', tbody, { class: 'label' });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend('div', container, {
+          text: error.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, { 'text': item.name, 'value': index });
+
+          renderRepositoryBox(
+            repositories[0],
+            repositoryLink,
+            description,
+            forks,
+            updated
+          );
+        });
+      }
+
+      select.addEventListener('change', () => {
+        container.innerHTML = "";
+        const index = event.target.value;
+
+        renderRepositoryBox(
+          repositories[select.value],
+          repositoryLink,
+          description,
+          forks,
+          updated
+        );
+        const contributorOnSelect = repositories[index].contributors_url;
+        fetchJSON(contributorOnSelect, (err, contributorData) => {
+          renderContributors(contributorData, container);
+        });
+      });
+      const contributorsInfo = repositories[0].contributors_url;
+      fetchJSON(contributorsInfo, (error, contributorData) => {
+        renderContributors(contributorData, container);
+      });
+
+      function renderRepositoryBox(
+        repositories,
+        repositoryLink,
```

Comment:
You do not have to pass the `repositoryLink`, `repositoryLink`, `description`, `forks` and `updated` since this are global variables:

You get them from 
```
const repositoryLink = createAndAppend('a', tr);
const description = createAndAppend('tr', tbody, { class: 'label' });
const forks = createAndAppend('tr', tbody, { class: 'label' });
const container = createAndAppend('div', root, { id: 'container' });
const updated = createAndAppend('tr', tbody, { class: 'label' });
```
Pass it to `renderRepositoryBox()` in: 
```
renderRepositoryBox(repositories[select.value], repositoryLink, description, forks, updated);
```
And use it in the body of `renderRepositoryBox()` for example on
```
repositoryLink.innerText = repositories.name;
forks.innerHTML = 'Forks: ' + repositories.forks;
description.innerHTML = 'Description: ' + repositories.description;
updated.innerHTML = 'Updated: ' + new Date(repositories.updated_at).toLocaleString();
```
If this were no arguments, you would still have access to the global variable, so 
```
// Declare global variables
const repositoryLink = createAndAppend('a', tr);
const description = createAndAppend('tr', tbody, { class: 'label' });
const forks = createAndAppend('tr', tbody, { class: 'label' });
const container = createAndAppend('div', root, { id: 'container' });
const updated = createAndAppend('tr', tbody, { class: 'label' });
// Declare function 
function renderRepositoryBox(repositories) { 
    // Use global variables in function, without them being arguments
    repositoryLink.innerText = repositories.name;
    forks.innerHTML = 'Forks: ' + repositories.forks;
    description.innerHTML = 'Description: ' + repositories.description; 
    updated.innerHTML = 'Updated: ' + new Date(repositories.updated_at).toLocaleString();
}

// Call the function without `repositoryLink`, `description`, `forks` and `updated` as arguments
renderRepositoryBox(repositories[select.value], description, forks, updated);
```

This is however because you have the function `renderRepositoryBox()` declared inside the function `main()`. That's why you have access to the variables. 
A better solution is to pass the values as an argument, but to a function that is outside (and above!) the `main()` function.